velox/dwio/dwrf/writer/ColumnWriter.cpp

```diff
@@ -697,6 +697,118 @@ class TimestampColumnWriter : public BaseColumnWriter {
   std::unique_ptr<IntEncoder<false>> nanos_;
 };
 
+class DecimalColumnWriter : public BaseColumnWriter {
+ public:
+  DecimalColumnWriter(
+      WriterContext& context,
+      const TypeWithId& type,
+      uint32_t sequence,
+      std::function<void(IndexBuilder&)> onRecordPosition)
+      : BaseColumnWriter{context, type, sequence, std::move(onRecordPosition)},
+        type_{type.type()},
+        scale_{getDecimalPrecisionScale(*type_).second},
+        isShortDecimal_{type_->isShortDecimal()},
+        unscaledValues_{createDirectEncoder<true /*isSigned*/>(
+            newStream(StreamKind::StreamKind_DATA),
+            // IntDecoder and IntEncoder only support vInts for huge ints.
+            isShortDecimal_ ? getConfig(Config::USE_VINTS) : true /*useVInts*/,
```

Comment:
It turns out the int decoder only supports vInts. And I removed the support for fixed-length and always use vInts for long decimal. @xiaoxmeng Do you think it makes sense? Thanks.

https://github.com/facebookincubator/velox/blob/8aeb51cf4443e7f7790217dd13d3bdae6f7c3b78/velox/dwio/common/DirectDecoder.cpp#L58-L60